gen/weather/weather.hoon

```diff
@@ -0,0 +1,52 @@
+/-  sole
+/+  generators
+=,  generators
+:-  %get
+|=  [* [city=tape country=tape appid=tape ~] ~]
+^-  (sole-request:sole (cask tang))
+=/  escaped-city  (en-urlt:html city)
+=/  escaped-country  (en-urlt:html country)
+%+  curl  (scan "http://api.openweathermap.org/data/2.5/weather?q={escaped-city}%2C{escaped-country}&APPID={appid}" auri:de-purl:html)
+|=  hit/httr:eyre
+?~  r.hit  !!
+=/  my-json  (de-json:html q.u.r.hit)
+?~  my-json  !!
+=,  dejs:format
+=/  my-map
+%-  (om same)
+u.my-json
+=/  name
+%.  %name
+%~  got  by
+my-map
+=/  weather
+%.  %weather
+%~  got  by
+my-map
+=/  main
+%.  %main
+%~  got  by
+my-map
+=/  wind
+%.  %wind
+%~  got  by
+my-map
+%+  produce  %tang
+=/  returned-name  ?~  name  ""  ?+  p.name  ""  @t  (scow %tas p.name)  ==
```

Comment:
This is long enough to split over multiple lines. Since it looks like `name` is a `json` noun, you can get rid of the `?~`, and do typechecking like this:

```
?+  name  ""
  [%s *]  (scow %tas p.name)
==